lib/install/validate/config_to_data.go

```diff
@@ -0,0 +1,314 @@
+	}
+	d.ClientCAs = conf.Certificate.CertificateAuthorities
+	d.RegistryCAs = conf.RegistryCertificateAuthorities
+
+	clientNet, err := getNetworkConfig(ctx, finder, conf.ExecutorConfig.Networks[config.ClientNetworkName])
+	if err != nil {
+		return
+	}
+	d.ClientNetwork = *clientNet
+	publicNet, err := getNetworkConfig(ctx, finder, conf.ExecutorConfig.Networks[config.PublicNetworkName])
+	if err != nil {
+		return
+	}
+	d.PublicNetwork = *publicNet
+	mgmtNet, err := getNetworkConfig(ctx, finder, conf.ExecutorConfig.Networks[config.ManagementNetworkName])
+	if err != nil {
+		return
+	}
+	d.ManagementNetwork = *mgmtNet
+
+	// remove duplicate network config
+	if d.ManagementNetwork.Name == d.ClientNetwork.Name {
+		d.ManagementNetwork = data.NetworkConfig{}
+	}
+	if d.ClientNetwork.Name == d.PublicNetwork.Name {
+		// revert client network settings
+		d.ClientNetwork = data.NetworkConfig{}
+	}
+
+	if err = setContainerNetworks(ctx, finder, d, conf.Network.ContainerNetworks, conf.BridgeNetwork); err != nil {
+		return
+	}
+
+	d.Debug.Debug = conf.Diagnostics.DebugLevel
+	if conf.ExecutorConfig.Networks[config.PublicNetworkName] != nil {
+		d.DNS = conf.ExecutorConfig.Networks[config.PublicNetworkName].Network.Nameservers
+	}
+
+	if err = setHTTPProxies(d, conf); err != nil {
+		return
+	}
+
+	if err = setImageStore(d, conf); err != nil {
+		return
+	}
+	setVolumeLocations(d, conf)
+	d.InsecureRegistries = conf.InsecureRegistries
+	if d.ScratchSize, err = getHumanSize(conf.ScratchSize, "KB"); err != nil {
+		return
+	}
+	if conf.Diagnostics.SysLogConfig != nil {
+		if d.SyslogConfig.Addr, err = url.Parse(fmt.Sprintf("%s://%s", conf.Diagnostics.SysLogConfig.Network, conf.Diagnostics.SysLogConfig.RAddr)); err != nil {
+			return
+		}
+	}
+	return
+}
+
+func getHumanSize(size int64, unit string) (string, error) {
+	is, err := units.FromHumanSize(fmt.Sprintf("%d%s", size, unit))
+	if err != nil {
+		return "", err
+	}
+	hsize := units.HumanSize(float64(is))
+	hsize = strings.Replace(hsize, " ", "", -1)
+	return hsize, nil
+}
+
+func setImageStore(d *data.Data, conf *config.VirtualContainerHostConfigSpec) error {
+	if len(conf.ImageStores) != 1 {
+		return fmt.Errorf("no image store or multiple image store configured")
+	}
+	imageURL := conf.ImageStores[0]
+	if imageURL.Path != "" {
+		path := strings.Split(imageURL.Path, "/")
+		if len(path) > 1 && path[len(path)-1] != "" {
+			imageURL.Path = strings.Join(path[:len(path)-1], "/")
+		}
+		if imageURL.Scheme != "" && len(path) == 1 {
+			imageURL.Path = ""
+		}
+	}
+	d.ImageDatastorePath = urlString(imageURL)
+	return nil
+}
+
+func setVolumeLocations(d *data.Data, conf *config.VirtualContainerHostConfigSpec) {
+	d.VolumeLocations = make(map[string]*url.URL, len(conf.VolumeLocations))
+
+	var dsURL object.DatastorePath
+	for k, v := range conf.VolumeLocations {
+		if ok := dsURL.FromString(v.Path); !ok {
+			log.Debugf("%s is not datastore path", v.Path)
```

Comment:
nfs volume store is not datastore path, so in that case, we set the datastore path directly